lib/anagrams.rb

```diff
@@ -0,0 +1,28 @@
+class Anagrams
+
+  def initialize(file)
+    @file = file
+    @result = {}
+  end
+
+  def read_wordlist
+    if @file.is_a?(Array)
+      return @file
+    else
+      File.read(@file).split("\n")
+    end
+  end
+
+  def get_anagrams
+    read_wordlist.each do |word|
+      key = word.split('').sort.join
+      if @result.key?(key)
```

Comment:
@StefanGajic the way you are using it now there's no need for `@result` to be an instance variable. It's only used as a local variable. If you wanted to cache the anagrams then you need to modify `get_anagrams` to actually use cached version instead of finding all anagrams on each call.

Reply:
thanks, I will just declare `result` as a local variable